src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/FlowStatsService.java

```diff
@@ -62,15 +63,16 @@ public void persistFlowStats(String flowId, String direction, long latency) {
                     if (flow.isPresent()) {
                         FlowStats toCreate = new FlowStats(flow.get(), null, null);
                         flowStatsRepository.add(toCreate);
-                        flowStats = toCreate;
-                    } else if (haSubFlowRepository.findById(flowId).isPresent()) {
-                        //TODO: https://github.com/telstra/open-kilda/issues/5223
-                    } else {
-                        log.warn("Can't save latency for flow '{}'. Flow not found.", flowId);
                         return;
                     }
-                }
-                if (flowStats != null) {
+                    Optional<HaSubFlow> haSubFlow = haSubFlowRepository.findById(flowId);
+                    if (haSubFlow.isPresent()) {
+                        FlowStats toCreate = new FlowStats(haSubFlow.get(), null, null);
+                        flowStatsRepository.add(toCreate);
+                        return;
```

Comment:
You are creating empty FlowStats (with null latency) and exit from the method. 
code `flowStats.setForwardLatency` will be never called and latency will have null values

Reply:
Good catch. Fixed. Thanks!